helpers/cache_helper.py

```diff
@@ -14,3 +14,14 @@ def __init__(self, cache_file):
     def save(self):
         with open(self.cache_file, "w") as f:
             json.dump(self.data, f)
+        self.data.dirty = False
+
+
+# Use this for the cache so we can mark it dirty on any update
+class CacheDict(dict):
+    def __init__(self):
+        self.dirty = False
+
+    def __setitem__(self, k, v):
```

Comment:
This the only place a `dict` can be updated? What about calls to `update()` or such things, does that all end up going through `__setitem__`?